internal/provisioner/kops_provisioner_cluster.go

```diff
@@ -1065,7 +1065,17 @@ func (provisioner *KopsProvisioner) prepareSloth(cluster *model.Cluster, k8sClie
 		},
 	}
 
-	err := k8sClient.CreateFromFiles(files)
+	podMonitorFile := k8s.ManifestFile{
+		Path:            "manifests/sloth/sloth_pod_monitor.yaml",
+		DeployNamespace: "prometheus",
+	}
+
+	err := k8sClient.DeleteFromFile(podMonitorFile)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete sloth pod monitor.")
+	}
```

Comment:
Can we perhaps try a different method of update by patching the resource instead of treating it as a special case? I see we do it for some resources https://github.com/mattermost/mattermost-cloud/blob/master/k8s/custom_resources.go#L66

I am not 100% sure that patching will work for this case and I am not sure if you already tried this but I would be 4/5 on doing it that way if possible.

Reply:
Worked thanks!